File: Programming EF Code First/chapter4/Chapter 4 VB/Data Annotations/Model/InternetSpecial.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  Public Class InternetSpecial
	Public Property InternetSpecialId() As Integer
	Public Property Nights() As Integer
	Public Property CostUSD() As Decimal
	Public Property FromDate() As Date
	Public Property ToDate() As Date

	<ForeignKey("Accommodation")>
	Public Property AccommodationId() As Integer
	Public Property Accommodation() As Lodging
  End Class
End Namespace

File: Programing EF DbContext/StartingSolution/Model/InternetSpecial.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Model
{
  public class InternetSpecial
  {
    public int InternetSpecialId { get; set; }
    public int Nights { get; set; }
    public decimal CostUSD { get; set; }
    public DateTime FromDate { get; set; }
    public DateTime ToDate { get; set; }

    [ForeignKey("Accommodation")]
    public int AccommodationId { get; set; }
    public Lodging Accommodation { get; set; }
  }
}